### PR TITLE
Make search better

### PR DESCRIPTION
**Pull request by rlgreen91**
This PR includes a number of changes relating to search, including:
- moving the search functionality to a separate controller
- refactoring the search functionality using a PORO, accounting for dynamic search terms and possible future expansion
- creating a separate search results page with customized text based on the search terms and filtering

Overall, this PR addresses issue #888 
----------------------------------------------------------------------------------------------------------------

In your PR did you:

  - [X] Include a description of the changes?
  - [X] Mention the issue the PR addresses?
  - [X] Include screenshots of any changes to the UI?
  - [ ] Isolate any changes to gems (meaning that any new, updated, or removed gems and resulting code changes should be in their own PR)?
  - [ ] Add and/or update specs for your code?

### CODE REVIEW

**Comment by rlgreen91** [comment]
search_results

File: app/assets/javascripts/search.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Place all the behaviors and hooks related to the matching controller here.
```
Comment (trystant) [inline app/assets/javascripts/search.js]: @rlgreen91 Can we remove this unneeded file?

File: app/assets/stylesheets/search.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Search controller here.
```
Comment (trystant) [inline app/assets/stylesheets/search.scss]: @rlgreen91 Can we remove this unneeded file?

File: app/helpers/search_helper.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
```
Comment (trystant) [inline app/helpers/search_helper.rb]: @rlgreen91 Can we remove this unneeded file?

File: spec/models/case_spec.rb [hunk 4]
```diff
@@ -213,22 +213,6 @@
   end
 end
 describe 'scopes', versioning: true do
-  it 'returns cases based on case' do
```
Comment (trystant) [inline spec/models/case_spec.rb]: @rlgreen91 Was this removed because it's testing Rails functionality (scopes)?

Comment (rlgreen91) [reply spec/models/case_spec.rb]: @trystant No, I removed it because I removed the scope - we no longer use it anywhere.

**Comment by StlMaris123** [comment]
Hi @rlgreen91 , I came across this resource that really helped to refactor my controllers on filter and search:  https://www.justinweiss.com/articles/search-and-filter-rails-models-without-bloating-your-controller/

**Comment by rlgreen91** [comment]
> Hi @rlgreen91 , I came across this resource that really helped to refactor my controllers on filter and search: https://www.justinweiss.com/articles/search-and-filter-rails-models-without-bloating-your-controller/

@StlMaris123 Yeah, I came across that a while ago.  However, that approach doesn't work as well with Searchkick.

**Pull request merged**